api/http/handlerfuncs.go

```diff
@@ -124,7 +126,7 @@ func execGQLHandler(rw http.ResponseWriter, req *http.Request) {
 				handleErr(req.Context(), rw, ErrBodyEmpty, http.StatusBadRequest)
 				return
 			}
-			body, err := io.ReadAll(req.Body)
+			body, err := readWithTimeout(req.Context(), req.Body, time.Second)
```

Comment:
in terms of concrete suggestion for my comments, this whole PR can be turned into a single line change, which also avoids context, additional goroutine for timeouts, select call etc.

```go
r := io.LimitReader(req.Body,  1 << (10 * 2)) // here  1 << (10 * 2) is a Megabyte, can be expressed however
body, err := io.ReadAll(r)
```

Reply:
`io.ReadAll` is potential very wasteful memory-wise as it relies on `append` for resizing the result slice. It could theoretically mean that the capacity (allocated mem cost) of the result is actually (2n - 1).

I also really dont want any more magic in the magic number, and doing that calc inline like that is not very readable, with or without a comment.